### PR TITLE
feat: update minimum supported DB versions for optimize and remove ha…

### PR DESCRIPTION
**Pull request by RomanJRW**
…rd checks on startup

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

https://github.com/camunda/camunda-optimize/issues/14041#event-13926325434

### CODE REVIEW

**Review by igpetrov — APPROVED** [review]
LGTM

**Review by HeleneW-dot — APPROVED** [review]
Lgtm, [these](https://docs.camunda.io/docs/reference/supported-environments/#component-requirements) docs also need updated to reflect this change plus a mention on supported environments changes in the upgrade notes for the next version

**Comment by RomanJRW** [comment]
@PHWaechtler - your link is to the old version of the docs, the [next ones](https://docs.camunda.io/docs/next/reference/supported-environments/#component-requirements) are already reflecting the correct versions

Good idea with the upgrade guide, I will do this when I return to work 👍

File: optimize/pom.xml [hunk 1]
```diff
@@ -26,17 +26,13 @@
     <surefire-plugin.version>3.4.0</surefire-plugin.version>
 
     <!-- database config -->
-    <elasticsearch.version>7.17.23</elasticsearch.version>
-    <opensearch.version>2.10.1</opensearch.version>
+    <elasticsearch.client.version>7.17.23</elasticsearch.client.version>
```
Comment (igpetrov) [inline optimize/pom.xml:29]: Out of curiosity: shouldn't it be 8.13.0? 🤔

**Review by igpetrov — APPROVED** [review]
Looked at the code again. Still looks good to me. Thank you!

File: optimize/pom.xml (hunk 1 above)
Comment (RomanJRW) [reply optimize/pom.xml:29]: This is just the client version, that doesn't get bumped up until Ivan's changes. This PR is just changing the support for the underlying DB

**Comment by RomanJRW** [comment]
For reference, documented in https://github.com/camunda/camunda-docs/pull/4192

**Pull request merged**